python/pyarrow/_s3fs.pyx

```diff
@@ -174,7 +174,8 @@ cdef class S3FileSystem(FileSystem):
         The frequency (in seconds) with which temporary credentials from an
         assumed role session will be refreshed.
     region : str, default 'us-east-1'
-        AWS region to connect to.
+        AWS region to connect to. Use :func:`pyarrow.fs.resolve_s3_region` to
```

Comment:
The default nowadays is not always "us-east-1", but will also depend on some configuration / env variables. See the expanded section in the C++ code about this: https://github.com/apache/arrow/blob/5889c78e344688f8fa8100ecdf254cd701ee3445/cpp/src/arrow/filesystem/s3fs.h#L105-L109 
That is maybe too detailed, but wondering if we should update it here a bit